pkg/agent/agent.go

```diff
@@ -1301,13 +1304,13 @@ func (i *Initializer) initNodeLocalConfig() error {
 func (i *Initializer) initVMLocalConfig(nodeName string) error {
 	var en *v1alpha1.ExternalNode
 	klog.InfoS("Initializing VM config", "ExternalNode", nodeName)
-	if err := wait.PollImmediateUntil(10*time.Second, func() (done bool, err error) {
+	if err := wait.PollUntilContextCancel(context.TODO(), 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
 		en, err = i.crdClient.CrdV1alpha1().ExternalNodes(i.externalNodeNamespace).Get(context.TODO(), nodeName, metav1.GetOptions{})
 		if err != nil {
 			return false, nil
 		}
 		return true, nil
-	}, i.stopCh); err != nil {
+	}); err != nil {
```

Comment:
Previously the loop would exit when stopCh is closed, the code changes it.

Reply:
Thanks for the reminder.